incubator/exporter-otlp/src/main/java/io/aklivity/zilla/runtime/exporter/otlp/internal/OltpExporterHandler.java

```diff
@@ -0,0 +1,175 @@
+import io.aklivity.zilla.runtime.exporter.otlp.internal.serializer.OtlpMetricsSerializer;
+
+public class OltpExporterHandler implements ExporterHandler
+{
+    private static final long RETRY_INTERVAL = Duration.ofSeconds(5).toMillis();
+    private static final long WARNING_INTERVAL = Duration.ofMinutes(5).toMillis();
+    private static final String DEFAULT_METRICS_URI = "/v1/metrics";
+    private static final String HTTP = "http";
+
+    private final EngineContext context;
+    private final Set<OtlpSignalsConfig.Signals> signals;
+    private final String protocol;
+    private final URI metricsUrl;
+    private final long interval;
+    private final Collector collector;
+    private final LongFunction<KindConfig> resolveKind;
+    private final List<AttributeConfig> attributes;
+    private final HttpClient httpClient;
+
+    private OtlpMetricsSerializer serializer;
+    private long lastAttempt;
+    private long lastSuccess;
+    private boolean warning;
+
+    public OltpExporterHandler(
+        EngineConfiguration config,
+        EngineContext context,
+        OtlpExporterConfig exporter,
+        Collector collector,
+        LongFunction<KindConfig> resolveKind,
+        List<AttributeConfig> attributes)
+    {
+        this.context = context;
+        this.metricsUrl = createMetricsUrl(exporter.options().endpoint);
+        this.signals = exporter.options().signals.signals;
+        this.protocol = exporter.options().endpoint.protocol;
+        this.interval = Duration.ofSeconds(exporter.options().interval).toMillis();
+        this.collector = collector;
+        this.resolveKind = resolveKind;
+        this.attributes = attributes;
+        this.httpClient = HttpClient.newBuilder().build();
+    }
+
+    @Override
+    public void start()
+    {
+        assert signals.contains(METRICS);
+        assert HTTP.equals(protocol);
+
+        MetricsReader metrics = new MetricsReader(collector, context::supplyLocalName);
+        serializer = new OtlpMetricsSerializer(metrics.records(), attributes, context::resolveMetric, resolveKind);
+        lastAttempt = 0;
+        lastSuccess = System.currentTimeMillis();
+        warning = false;
+    }
+
+    @Override
+    public int export()
+    {
+        long now = System.currentTimeMillis();
+        if (now - lastSuccess < interval)
+        {
+            return 0;
+        }
+        if (now - lastAttempt < RETRY_INTERVAL)
+        {
+            return 0;
+        }
+        if (now - lastSuccess >= WARNING_INTERVAL && !warning)
+        {
+            System.out.format("Warning: Could not successfully publish data to OpenTelemetry Collector for %d seconds.%n",
+                Duration.ofMillis(now - lastSuccess).toSeconds());
+            warning = true;
+        }
+        post();
+        return 1;
+    }
+
+    // required for testing
+    URI metricsUrl()
+    {
+        return metricsUrl;
+    }
```

Comment:
I think we can hopefully remove this after the `OtlpExporterEndpointConfig` changes.